persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -702,14 +702,22 @@ private Cursor loadEventsCursor(final long measurementIdentifier) {
                 // Pause reached: Move geoLocationCursor to the first location of the next sub-track
                 // We do this to ignore locations between pause and resume event (STAD-140)
                 while (location.getTimestamp() < resumeEventTime) {
-                    geoLocationCursor.moveToNext();
+                    if (!geoLocationCursor.moveToNext()) {
```

Comment:
You should probably use hasNext here, since the while loop is responsible for the moveToNext. 

Why is the if condition not part of the while loop condition? It gets called on each iteration at first, which is exactly what the while loop does, isn't it?

Reply:
Thought it's better readable but it's changed now.

Reply:
There is no "hasNext" for cursors. Thus, moveToNext() is used. But see refactored code for an alternative implementation.